Chapter 8/routes/todos/autohooks.js

```diff
@@ -2,31 +2,25 @@
 const fp = require('fastify-plugin')
 const schemas = require('./schemas/loader')
 
-module.exports = fp(todoAutoHooks, {
-  dependencies: ['@fastify/mongodb'],
-  name: 'todo-store'
-})
-
-async function todoAutoHooks (fastify, opts) {
+module.exports = fp(async function todoAutoHooks (fastify, opts) {
   const todos = fastify.mongo.db.collection('todos')
 
   fastify.register(schemas)
 
-  fastify.decorateRequest('mongoDataSource', function () {
-    const req = this
-    return {
+  fastify.decorateRequest('todosDataSource', null) // [1]
```

Comment:
I used this because of what we recommend in the official docs about security and memory leaks https://www.fastify.io/docs/latest/Reference/Decorators/#decoraterequestname-value-dependencies